### PR TITLE
Fix decimalPrecision with custom decimalSeparator

### PR DESCRIPTION
**Pull request by alexandrapersea**
This PR fixes [issue #37](https://github.com/s-yadav/react-number-format/issues/37)

### CODE REVIEW

File: example/src/index.js [hunk 1]
```diff
@@ -75,6 +75,25 @@ class App extends React.Component {
 
         <div className="example">
           <h3>
+            Custom thousand separator with decimal precision
+          </h3>
+          <div>
+            ThousandSeperator: ',', decimalSeparator='.', decimalPrecision:2
+          </div>
+          <div>
+            <NumberFormat thousandSeparator={","} decimalSeparator={'.'} decimalPrecision={2} />
```
Comment (tudorpavel) [inline example/src/index.js]: You should keep a consistent style for string quotes.

File: test/input_test.js [hunk 2]
```diff
@@ -91,6 +91,17 @@ describe('NumberFormat as input', () => {
     input.simulate('change', getCustomEvent("2456981'89"));
     expect(wrapper.state().value).toEqual("$2 456 981'89");
 
+    wrapper.setProps({thousandSeparator: ",", decimalSeparator: "."});
+    input.simulate('change', getCustomEvent("2456981.89"));
+    expect(wrapper.state().value).toEqual("$2,456,981.89");
+  });
+
+  it('should support decimal precision with custom decimal separator', () => {
+    const wrapper = shallow(<NumberFormat thousandSeparator={"."} decimalSeparator={','} decimalPrecision={2} />);
```
Comment (tudorpavel) [inline test/input_test.js]: You should keep a consistent style for string quotes.

File: test/input_test.js [hunk 3]
```diff
@@ -91,6 +91,17 @@ describe('NumberFormat as input', () => {
...
+    const input = wrapper.find('input');
+
+    input.simulate('change', getCustomEvent('2456981,89'));
```
Comment (tudorpavel) [inline test/input_test.js]: Nice, did you consider adding test cases for when given shorter/longer precision?
- `2456981,894` --> `2456981,89`
- `2456981,8` --> `2456981,80`

**Comment by alexandrapersea** [comment]
@s-yadav Could you review this PR?

**Comment by WagnerSilveira** [comment]
@s-yadav could you merge this correction, please? :D 
we need it too much !
Thanks

File: lib/number_format.js [hunk 4]
```diff
@@ -190,7 +190,15 @@ var NumberFormat = function (_React$Component) {
           var parts = void 0;
           if (decimalPrecision !== false) {
             var precision = decimalPrecision === true ? 2 : decimalPrecision;
-            parts = parseFloat(formattedValue).toFixed(precision).split(decimalSeparator);
+            if (decimalSeparator === ',') {
```
Comment (s-yadav) [inline lib/number_format.js]: Can you change it to 
```js
if (decimalSeparator === ',') {
   parts = parseFloat(formattedValue.replace(decimalSeparator, '.')).toFixed(precision);
  ....
```
As there can be other seperator as well like ' or : .

**Comment by alexandrapersea** [comment]
@s-yadav Comments fixed.

**Pull request merged**